test/Dialect/FIRRTL/SFCTests/data-taps-more-xmrs.fir

```diff
@@ -233,26 +233,26 @@ circuit Top : %[[
     wire tap : UInt<1>[6]
     tap is invalid
 
-    ; CHECK:      module Submodule
-    ; CHECK:      wire tap_3 = 1'h0
-    ; CHECK-NEXT: wire tap_4 = 1'h0
-    ; CHECK:      assign tap_0 = Submodule.wire_Submodule
-    ; CHECK-NEXT: assign tap_1 = DUT.wire_DUT
-    ; CHECK-NEXT: assign tap_2 = Top.wire_Top
-    ; CHECK-NEXT: assign tap_5 = Top.port_Top
+    ; CHECK-LABEL: module Submodule
+    ; CHECK:         wire tap_3 = 1'h0
+    ; CHECK-NEXT:    wire tap_4 = 1'h0
+    ; CHECK-DAG:     assign tap_0 = wire_Submodule
+    ; CHECK-DAG:     assign tap_1 = DUT.wire_DUT
+    ; CHECK-DAG:     assign tap_2 = Top.wire_Top
+    ; CHECK-DAG:     assign tap_5 = Top.port_Top
 
-    ; CHECK: module DUT
-    ; CHECK:      wire tap_3 = 1'h0
-    ; CHECK-NEXT: wire tap_4 = 1'h0
-    ; CHECK:      assign tap_0 = DUT.submodule.wire_Submodule
-    ; CHECK-NEXT: assign tap_1 = DUT.wire_DUT
-    ; CHECK-NEXT: assign tap_2 = Top.wire_Top
-    ; CHECK-NEXT: assign tap_5 = Top.port_Top
+    ; CHECK-LABEL: module DUT
+    ; CHECK:         wire tap_3 = 1'h0
+    ; CHECK-NEXT:    wire tap_4 = 1'h0
+    ; CHECK-DAG:     assign tap_0 = DUT.submodule.wire_Submodule
+    ; CHECK-DAG:     assign tap_1 = wire_DUT
+    ; CHECK-DAG:     assign tap_2 = Top.wire_Top
+    ; CHECK-DAG:     assign tap_5 = Top.port_Top
 
-    ; CHECK: module Top
-    ; CHECK:      wire tap_3 = 1'h0
-    ; CHECK-NEXT: wire tap_4 = 1'h0
-    ; CHECK:      assign tap_0 = Top.dut.submodule.wire_Submodule
-    ; CHECK-NEXT: assign tap_1 = Top.dut.wire_DUT
-    ; CHECK-NEXT: assign tap_2 = Top.wire_Top
-    ; CHECK-NEXT: assign tap_5 = Top.port_Top
+    ; CHECK-LABEL: module Top
+    ; CHECK:         wire tap_3 = 1'h0
+    ; CHECK-NEXT:    wire tap_4 = 1'h0
+    ; CHECK-NEXT:    wire tap_5 = 1'h0
```

Comment:
This seems to be correct, `Top.port_Top` is `invalid` and un-assigned.

Reply:
In short, yes.  It's "correct" but wasn't sure why happened now (and why other taps don't get it, if IMCP is doing this).
Poked at this a bit and it's because we don't IMCP through public output ports, but in HW canonicalize propagates the zero to the wires (within `Top`).  This is why the zero doesn't reach the other taps of the port (`port_Top`) since at that point the ref paths are opaque verbatim's.

Reply:
FWIW getting things wrong in some ways (e.g., re:where connect's are inserted) can look like this too -- since the propagated value is same as the non-tapped value they're given (invalid).